src/world/Map/MapScriptInterface.cpp

```diff
@@ -185,7 +185,7 @@ Creature* MapScriptInterface::SpawnCreature(MySQLStructure::CreatureSpawn* sp, b
     uint8 Gender = creature_properties->GetGenderAndCreateRandomDisplayID(&sp->displayid);
     Creature* p = this->mapMgr.CreateCreature(sp->entry);
     ARCEMU_ASSERT(p != NULL);
-    p->Load(sp, static_cast<unsigned int>(reinterpret_cast<uintptr_t>(NULL)), nullptr);
+    p->Load(sp, reinterpret_cast<uintptr_t>(NULL), nullptr);
```

Comment:
You could remove all casts and replace the null with 0.

Reply:
yes i know, this option i use when i surrender